### ISSUE TITLE
F# compiler service does not run unresolved references through metadata resolver

### ISSUE THREAD

**Issue by fabiocav** [Azure/azure-webjobs-sdk-script#819]
**NOTE:** Opening this to document the issue since @dsyme has some thoughts on the appropriate way to collect the unresolved/referenced assemblies.

Currently, the F# compiler service does not run unresolved assembly references through the metadata resolver, like C#, so shared assemblies and extension provided assemblies are not resolved.
#### Repro steps
1. Create a simple manual trigger F# function
2. Add a #r reference to "Twilio.Api"

The Twilio assembly will not be resolved.

Here's the example function:

``` fsharp
#r "Twilio.Api"

let Run(input: string, log: TraceWriter) =
    log.Info(
        sprintf "F# manually triggered function called with input: %s" input)

```
#### Expected behavior

Shared assemblies are properly resolved.
#### Actual behavior

Unable to used shared assemblies.
#### Related information

Running the references through the metadata resolver would get the shared assemblies (https://github.com/fabiocav/azure-webjobs-sdk-script/blob/480924c2908fd7acb84ad23e38d219e4682bf630/src/WebJobs.Script/Description/DotNet/FunctionMetadataResolver.cs#L215)
- F#

**Comment by fabiocav** [comment]
@dsyme, this is the issue I brought up. Please let me know if you need more information.

**Comment by dsyme** [comment]
@fabiocav We can fix this for resolution by the AF implementation.

However one major concern I have here is about the local in-editor editing experience.  How does the editor find ``Twilio.Api``?  See for example the "Editor Prelude" section here: https://azure.microsoft.com/en-us/documentation/articles/functions-reference-fsharp/

**Comment by fabiocav** [comment]
That is indeed a bit of a challenge and something we had to balance in order to enable the experience where bindings would be able to make those assemblies available to the function by default, without requiring an additional gesture.

There are some ideas on how to address the issue and it will become a bit simpler once we move forward with the extensibility plans, as extensions would be able to bring additional references as package dependencies, but for now, we've been handling (in a limited fashion) with a bit of tooling work (CLI and some work on OmniSharp) and documentation.

**Comment by dsyme** [comment]
@lindydonna @fabiocav @christopheranderson - @sylvanc and I are actively looking into this bug this week. We think nearly all the existing F# issues boil down to the one same bug (not finding assemblies at runtime which have been resolved statically)

Suspected duplicate of #650, but we need to investigate further.

**Comment by fabiocav** [comment]
Resolved by #893
